docs/API.md

```diff
@@ -0,0 +1,267 @@
+
+The `beforeInit` phase has no default behavior.  It can be used to perform actions prior to any of the other phases, but after `App.initialize` has validated its environment configuration.  If you want to perform actions prior to validation of the environment configuration, then write your code before calling `App.initialize` itself.
+
+### configuration (`APP_CONFIGURED`)
+
+The `configuration` phase has no default behavior.
+
+The `configuration` phase can be used to provide dynamic, runtime configuration prior to the initialization of any other services the application may need.
+
+### logging (`APP_LOGGING_CONFIGURED`)
+
+The `logging` phase initializes the NewRelicLoggingService from @edx/frontend-logging by default.
+
+### authentication (`APP_AUTHENTICATED`)
+
+The `authentication` phase creates an authenticated apiClient and makes it available at `App.apiClient` on the `App` singleton.  It also runs `ensureAuthenticatedUser` from @edx/frontend-auth and will redirect to the login experience if the user does not have a valid authentication cookie.  Finally, it will make authenticated user information available at `App.authenticatedUser` and `App.decodedAccessToken` for later use by the application.
+
+### i18n (`APP_I18N_CONFIGURED`)
+
+The `i18n` phase initializes @edx/frontend-i18n with the `messages` object provided to `App.initialize`.
+
+### analytics (`APP_ANALYTICS_CONFIGURED`)
+
+The `analytics` phase initializes Segment and configures @edx/frontend-analytics.
+
+### beforeReady (`APP_BEFORE_READY`)
+
+The `beforeReady` phase has no default behavior.
+
+### ready (`APP_READY`)
+
+The `ready` phase has no default behavior.  This is the phase where an application's interface would generally be shown to the user.
+
+### error (`APP_ERROR`)
+
+The `error` phase logs (to loggingService) whatever error occurred to put the app in an error state.  This is the phase where an application would generally show an error message for an unexpected error to the user.
+
+Note that the error which caused the application to transition to the `error` phase is available at `App.error`.  It is also passed as data to any subscribers to the `APP_ERROR` event.
+
+## `App`
+
+The `App` class is a singleton with static methods.  This is so that it can be used throughout a consuming application merely by importing it:
+
+```
+import { App } from '@edx/frontend-base';
+```
+
+`App` provides a few different methods and properties:
+
+### `App.history`
+
+The application's browser history object.  See the [history documentation](https://github.com/ReactTraining/history/blob/master/docs/GettingStarted.md) for more information.
+
+### `App.authenticatedUser`
+
+A reference to an object containing information about the currently authenticated user.
+
+```
+{
+  userId: <AUTHENTICATED USER's USER ID>,
+  username: <AUTHENTICATED USER's USERNAME>,
+  roles: [<ROLE ONE>, <ROLE TWO>],
+  administrator: <true|false>,
+}
+```
+### `App.decodedAccessToken`
```

Comment:
I wonder whether we should note that this is available for debugging purposes, but that `authenticatedUser` represents the contract.  Note that we've had thoughts in the past about cleaning up the access token, and exposing it makes that more difficult or impossible.  Maybe @nasthagiri has thoughts on this?

Reply:
Thanks for this - I've clarified along these lines.